playground/infrastructure/helper.py

```diff
@@ -44,29 +58,194 @@ def find_examples(work_dir: str) -> List[Example]:
     Update example.status with resulting status.
 
     Args:
         examples: beam examples for processing and updating statuses.
     """
     # TODO [BEAM-13267] Implement
     pass
+
+
+def get_tag(filepath):
+    """
+    Parse file by filepath and find beam tag
+
+    Args:
+        filepath: path of the file
+
+    Returns:
+        If file contains tag, returns tag as a map.
+        If file doesn't contain tag, returns None
+    """
+    add_to_yaml = False
+    yaml_string = ""
+
+    with open(filepath) as parsed_file:
+        lines = parsed_file.readlines()
+
+    for line in lines:
+        line = line.replace("// ", "").replace("# ", "")
+        if add_to_yaml is False:
+            if line == BEAM_PLAYGROUND_TITLE:
+                add_to_yaml = True
+                yaml_string += line
+        else:
+            yaml_with_new_string = yaml_string + line
+            try:
+                yaml.load(yaml_with_new_string, Loader=yaml.SafeLoader)
+                yaml_string += line
+            except YAMLError:
+                break
+
+    if add_to_yaml:
+        tag_object = yaml.load(yaml_string, Loader=yaml.SafeLoader)
+        return tag_object[BEAM_PLAYGROUND]
+
+    return None
+
+
+def _get_supported_categories(categories_path: str) -> List[str]:
+    """
+    Return list of supported categories from categories_path file
+
+    Args:
+        categories_path: path to the file with categories.
+
+    Returns:
+        All supported categories as a list.
+    """
+    with open(categories_path) as supported_categories:
+        yaml_object = yaml.load(supported_categories.read(), Loader=yaml.SafeLoader)
+        return yaml_object[TagFields.CATEGORIES]
+
+
+def _get_example(filepath: str, filename: str, tag: dict) -> Example:
+    """
+    Return an Example by filepath and filename.
+
+    Args:
+         tag: tag of the example.
+         filepath: path of the example's file.
+         filename: name of the example's file.
+
+    Returns:
+        Parsed Example object.
+    """
+    name = _get_name(filename)
+    sdk = _get_sdk(filename)
+    with open(filepath) as parsed_file:
+        content = parsed_file.read()
+
+    return Example(name, "", sdk, filepath, content, "", STATUS_UNSPECIFIED, Tag(**tag))
+
+
+def _validate(tag: dict, supported_categories: List[str]) -> bool:
+    """
+    Validate all tag's fields
+
+    Validate that tag contains all required fields and all fields have required format.
+
+    Args:
+        tag: beam tag to validate.
+        supported_categories: list of supported categories.
+
+    Returns:
+        In case tag is valid, True
+        In case tag is not valid, False
+    """
+    valid = True
+    for field_name in TAG_FIELDS:
+        if tag.get(field_name) is None:
+            logging.error("tag doesn't contain " + field_name + " field: " + tag.__str__())
+            valid = False
+
+    multifile = tag.get(TagFields.MULTIFILE)
```

Comment:
Yes. But as I said before this method verifies that the received dict is a correct tag. This dict may contain different types for some fields (`multifile`/`categories`)

Reply:
Btw, for namedtuple it is not strict to have boolean at multifile or list to categories. It will put anythint to Tag object if it has appropriate field names.

Reply:
Also, method `get_tag` could return a dict that doesn't contain all required fields (has only `"name": "Beam example"`, for example). In that case, as I can understand if I return:
```
tag_object = yaml.load(yaml_string, Loader=yaml.SafeLoader)
return Tag(**tag_object[BEAM_PLAYGROUND])
```
then I will receive an error:
![image](https://user-images.githubusercontent.com/26118465/143256399-3f5758a5-2623-4775-a4ab-efad75406e86.png)

Reply:
Right, if fields are missing then an error occurs